core/src/main/scala/org/bitcoins/core/protocol/script/ScriptPubKey.scala

```diff
@@ -545,6 +545,82 @@ object CSVScriptPubKey extends ScriptFactory[CSVScriptPubKey] {
 
 }
 
+/** Currently only supports a single OP_IF ... OP_ELSE ... OP_ENDIF ScriptPubKey */
+sealed trait ConditionalScriptPubKey extends ScriptPubKey {
+  require(asm.nonEmpty, "ConditionalScriptPubKey cannot be empty")
+  require(asm.head.equals(OP_IF),
```

Comment:
normally i would say skip it to simplify for now, but it is used in Lightning scripts:

https://github.com/lightningnetwork/lightning-rfc/blob/master/03-transactions.md#offered-htlc-outputs

Reply:
Another consideration: 

What if the condition is _not_ the first thing evaluated in the Script? Is that still a conditional Script? I.e. 

https://github.com/lightningnetwork/lightning-rfc/blob/master/03-transactions.md#offered-htlc-outputs

Reply:
Finally, I know you have the stated goal of not fully supporting conditionals here, which I think is wise. I just want to provide an example of a valid conditional Script that shows the insanity of Script, so we can at least mull over what it would look like to support nesting: https://github.com/bitcoin-s/bitcoin-s/blob/master/core-test/src/test/resources/script_tests.json#L62

Reply:
So I'm leaning towards getting this in and then adding an `OP_NOTIF` immediately after which should be much easier once `OP_IF` is in, at the very least I plan on getting all of the `OP_IF` code done on this PR before adding any `OP_NOTIF` related commits.

Re `OP_IF` not being first if this is the case I would say that good design would dictate a new SPK that has a `ConditionalScriptPubKey` nested within it, much like how my `MultiSignatureWithTimeoutScriptPubKey` will be nested within a `ConditionalScriptPubKey`. As a currently supported example, if a both conditions have the same locktime then one should construct a `CLTVScriptPubKey(ConditionalScriptPubKey(...`

That being said, I am aware that this will not work for nesting `OP_IF` scripts since my current asm processing code assumes a single `OP_IF ... OP_ELSE ... OP_ENDIF` but I will improve on this in the future.

Reply:
Also Re BOLT 3 offered HTLCs, those would require nesting anyway

Reply:
Something that might be helpful with understanding nesting is looking at the implementation of OP_IF in the interpreter, we use a binary tree like data structure that represents the program. 

https://github.com/bitcoin-s/bitcoin-s/blob/master/core/src/main/scala/org/bitcoins/core/script/control/ControlOperationsInterpreter.scala#L22

Tangentially, we will probably need to integrate something like this for Taproot, so we might as well start thinking about the best way to handle this generically.